kw/platform/requests_utils.py

```diff
@@ -0,0 +1,60 @@
+import os
+
+import wrapt
+
+
+APP_NAME = os.getenv("APP_NAME")
+PACKAGE_VERSION = os.getenv("PACKAGE_VERSION", "1.0.0")
+ENVIRONMENT = os.getenv("ENVIRONMENT", "dev")
+
+if APP_NAME:
+    USER_AGENT = "{app_name}/{package_version} (Kiwi.com {environment})".format(
+        app_name=APP_NAME, package_version=PACKAGE_VERSION, environment=ENVIRONMENT
+    )
+else:
+    USER_AGENT = None
+
+
+def set_user_agent(app_name=None, package_version=None, environment=None):
+    """Allows to customize or set User-Agent manually."""
+    app_name = app_name or APP_NAME
+    package_version = package_version or PACKAGE_VERSION
+    environment = environment or ENVIRONMENT
+    global USER_AGENT
+    if app_name:
+        USER_AGENT = "{app_name}/{package_version} (Kiwi.com {environment})".format(
+            app_name=APP_NAME, package_version=PACKAGE_VERSION, environment=ENVIRONMENT
+        )
+    else:
+        USER_AGENT = None
+
+
+def _add_user_agent(user_agent):
+    def _add_headers(func, instance, args, kwargs):
+        headers = kwargs.setdefault("headers", {})
+        if not headers.get("User-Agent"):
+            # If header 'User-Agent' has not been provided request needs to be patched.
+            if not user_agent:
+                # If no user_agent nor APP_NAME has been provided, request can't be
+                # patched.
+                raise ValueError(
+                    "Unable to patch requests 'User-Agent' header. You have to provide"
+                    " either 'APP_NAME' environment variable or patch manually using "
+                    "function patch_with_user_agent(user_agent=None) and provide "
+                    "'user_agent' argument."
+                )
+            else:
+                headers["User-Agent"] = user_agent
+        return func(*args, **kwargs)
+
+    return _add_headers
+
+
+def patch_with_user_agent(user_agent=None):
```

Comment:
Is your plan here to call each patcher one by one? Just want to clarify. So we'd later have

```
patch_with_user_agent()
patch_with_deprecation_warnings()
```

next to each other?